### ISSUE TITLE
WARNING:absl:Failed to converge after 99 steps: err_norm=0.0107458

### ISSUE THREAD

**Issue by ChenyangRan** [deepmind/dm_control#151]
HI, when I using the default reacher_easy env to train the sac agent, the error raises.
It seems that the code can work but I don't what happended. Is there anyone meet the same error?
```
WARNING:absl:Failed to converge after 99 steps: err_norm=0.0107458
```

**Comment by alimuldal** [comment]
TL;DR: this warning is not something you usually need to worry about.

During episode initialization we sample a random initial position for the end effector of the arm, and then use [inverse kinematics](https://en.wikipedia.org/wiki/Inverse_kinematics) (IK) to try to find a configuration of the arm joints that places the end effector at the desired position and orientation. Sometimes IK will fail to converge on the desired position, in which case [it will log the warning that you're seeing](https://github.com/deepmind/dm_control/blob/master/dm_control/utils/inverse_kinematics.py#L222).

However this is not normally a problem, since we [repeatedly sample random initial positions for the end effector until we find one for which IK converges (and that doesn't result in collisions)](https://github.com/deepmind/dm_control/blob/master/dm_control/composer/initializers/tcp_initializer.py#L154). If we fail to find one within 10 attempts then we raise a `RuntimeError`. In theory this could happen if you're very unlucky, but in practice such failures are extremely rare.
